verifier/verifier_test.go

```diff
@@ -605,3 +605,109 @@ func TestVerifyX509TrustedIdentities(t *testing.T) {
 		})
 	}
 }
+
+func TestPluginVersionCompatibility(t *testing.T) {
+	var semverSigEnv1 []byte
+	var semverSigEnv2 []byte
+	var semverSigEnv3 []byte
+	var semverSigEnv4 []byte
+	var semverSigEnv5 []byte
+
+	semverSigEnv1 = mock.MockSemVerSigEnv1
+	semverSigEnv2 = mock.MockSemVerSigEnv2
+	semverSigEnv3 = mock.MockSemVerSigEnv3
+	semverSigEnv4 = mock.MockSemVerSigEnv4
+	semverSigEnv5 = mock.MockSemVerSigEnv5
+
+	policyDocument := dummyPolicyDocument()
+
+	//envelopeMediaType := opts.SignatureMediaType
+	v := verifier{
+		trustPolicyDoc: &policyDocument,
+		pluginManager:  mock.PluginManager{},
+	}
+	opts := notation.VerifyOptions{ArtifactReference: mock.SampleArtifactUri, SignatureMediaType: "application/jose+json"}
+	envelopeMediaType := opts.SignatureMediaType
+	trustPolicy, _ := v.trustPolicyDoc.GetApplicableTrustPolicy(mock.SampleArtifactUri)
+	pluginConfig := opts.PluginConfig
+	verificationLevel, _ := trustPolicy.SignatureVerification.GetVerificationLevel()
+	outcome1 := &notation.VerificationOutcome{
+		RawSignature:      semverSigEnv1,
+		VerificationLevel: verificationLevel,
+	}
+	//first test case signature with minimum verification plugin version 1.0.1
+	err1 := v.processSignature(context.Background(), semverSigEnv1, envelopeMediaType, trustPolicy, pluginConfig, outcome1)
+	outcome1.Error = err1
```

Comment:
why do we need to store it in a variable?

Reply:
Modified the code accordingly